ui/src/views/user/settings/SettingsView.vue

```diff
@@ -144,28 +145,47 @@ export default {
       return this.userData.email;
     },
     isPasswordLengthOkay() {
-      return this.formData.password.length >= 8;
+      return this.formData.new_password.length >= 8;
     },
   },
   methods: {
     onFormSubmit() {
-      // to do: post form data to back-end's update user api
-      this.$notify.toast("New settings have been saved!");
+      userChangePassword(this.formData).then((response) => {
+        this.$notify.toast(response["message"]);
+      });
     },
     onDeleteAccount() {
       this.isDeleteItemModalShown = true;
     },
     onDeleteConfirmation() {
-      // to do: call delete api
-      this.isDeleteItemModalShown = false;
-      this.$store.dispatch("user/setToken", "");
-      this.$store.dispatch("user/setUserData", {});
-      this.$store.dispatch("user/setItems", []);
-      this.$cookies.remove("access_token_cookie");
-
-      this.$router
-        .push({ name: "general-user-delete-thankyou" })
-        .catch(() => {});
+      // first, call the user delete api
+      userDeleteAccount().then((response) => {
+        this.$notify.toast(response["message"]);
+        // then, call the user logout api
+        userLogout()
+          .then(() => {
+            // finally, unset client-side data and redirect to thankyou page
+            this.isDeleteItemModalShown = false;
```

Comment:
rather than copying the same lines twice, you can make a new function called `clearUserData` and call it in both blocks; that way we can ensure both blocks call the same function and its easier to read as well XD

Reply:
ok I will refactor it, I was trying to finish this asap that it escaped my attention that I should take that code out into its own function